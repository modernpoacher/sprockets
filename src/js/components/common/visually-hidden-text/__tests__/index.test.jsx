import React from 'react'
import {
  snapshotOf
} from 'react-component-snapshot'

import '@testing-library/jest-dom'

import {
  render
} from '@testing-library/react'

import VisuallyHiddenText from '#sprockets/components/common/visually-hidden-text'

describe('#sprockets/components/common/visually-hidden-text', () => {
  describe('<VisuallyHiddenText />', () => {
    it('renders', () => {
      const {
        container: {
          firstElementChild: visuallyHiddenText
        }
      } = render(
        <VisuallyHiddenText
          visuallyHiddenText='MOCK VISUALLY HIDDEN TEXT'
        />
      )

      expect(visuallyHiddenText)
        .toBeInstanceOf(HTMLSpanElement)
    })

    it('matches the snapshot', () => {
      const {
        container: {
          firstElementChild: visuallyHiddenText
        }
      } = render(
        <VisuallyHiddenText
          visuallyHiddenText='MOCK VISUALLY HIDDEN TEXT'
        />
      )

      expect(snapshotOf(visuallyHiddenText))
        .toMatchSnapshot()
    })
  })
})
