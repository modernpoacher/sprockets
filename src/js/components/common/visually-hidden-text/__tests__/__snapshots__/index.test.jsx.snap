// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`#sprockets/components/common/visually-hidden-text <VisuallyHiddenText /> matches the snapshot 1`] = `
<span
  className="visually-hidden-text"
>
  MOCK VISUALLY HIDDEN TEXT
</span>
`;
